fearless/Modules/Staking/SelectValidatorsFlow/ValidatorInfo/Interactor/YourValidatorInfoInteractor.swift

```
@@ -0,0 +1,84 @@
+            return
+        }
+
+        let operation = validatorOperationFactory.wannabeValidatorsOperation(for: [accountId])
+
+        operation.targetOperation.completionBlock = {
+            DispatchQueue.main.async { [weak self] in
+                do {
+                    if let validatorInfo =
+                        try operation.targetOperation.extractNoCancellableResultData().first {
+                        self?.presenter.didReceiveValidatorInfo(result: .success(validatorInfo))
+                    } else {
+                        let validatorInfo = SelectedValidatorInfo(address: self?.accountAddress ?? "")
+                        self?.presenter.didReceiveValidatorInfo(result: .success(validatorInfo))
+                    }
+                } catch {
+                    self?.presenter.didReceiveValidatorInfo(result: .failure(error))
+                }
+            }
+        }
+
+        operationManager.enqueue(operations: operation.allOperations, in: .transient)
+    }
+
+    private func fetchValidatorInfo() {
+        let operation = validatorOperationFactory.allElectedOperation()
+
+        operation.targetOperation.completionBlock = {
+            DispatchQueue.main.async { [weak self] in
+                do {
+                    let electedValidatorList = try operation.targetOperation.extractNoCancellableResultData()
+                    let searchValidatorResult = electedValidatorList.filter { validatorInfo in
+                        validatorInfo.address == self?.accountAddress
+                    }
+
+                    if let electedValidatorInfo = searchValidatorResult.first {
+                        let validatorInfo = electedValidatorInfo.toSelected(for: nil)
+                        self?.presenter.didReceiveValidatorInfo(result: .success(validatorInfo))
+                    } else {
+                        self?.fetchWannabeValidatorInfo()
+                    }
+                } catch {
```

Imagine a situation when error is not handled here, we're calling fetchWannabeValidatorInfo(), but 
```
guard let accountId = try? addressFactory.accountId(from: accountAddress) else {
            return
        }
```
accountId is null, so the UI state remains inconsistent